plugins/blip/basics/logic/datamunger.js

```diff
@@ -28,6 +28,42 @@ module.exports = function(bgClasses) {
   var classifiers = classifiersMkr(bgClasses);
 
   return {
+    calculateCarbStats: function(basicsData) {
+
+      var wizardEvents = basicsData.data.wizard.data;
+
+      var carbs = _.filter(wizardEvents, function(wizardEvent) {
+        return wizardEvent.carbInput && wizardEvent.carbInput > 0 ;
+      });
+
+      if (!_.isEmpty(carbs)) {
+
+        var start = carbs[0].normalTime;
+        if (start < basicsData.dateRange[0]) {
+          start = basicsData.dateRange[0];
+        }
+        var end = carbs[carbs.length - 1].normalTime;
+        if (end > basicsData.dateRange[1]) {
+          end = basicsData.dateRange[1];
+        }
```

Comment:
The adjustment of `start` and `end` here is unnecessary - it'd be nice if you could remove it. It's only necessary for `basal`s, which are interval events rather than point-in-time, so when we filter for all the `basal`s that fall within the basics date range, we include those that *overlap* with the range, which means that we then have to trim off the overlap. We can and are more precise with the filtering of other data types down to the determined date range for the basics.